### PR TITLE
SAK-34477: Let user configure custom delimiters in FIB questions.

### PR DESCRIPTION
**Pull request by FerAlvarez95**

### CODE REVIEW

**Comment by FerAlvarez95** [comment]
Added unit tests and some little changes to https://github.com/sakaiproject/sakai/pull/8619

**Comment by FerAlvarez95** [comment]
The main changes from the previous PR are:
 - Extract parsing methods into GradingService.java from ItemHelper12Impl.java
 - Extract validating methods into GradingService.java from ItemAddListenner.java
 - Add into GradingServiceTest.java unit test for parsing and validating methods

**Comment by ottenhoff** [comment]
Thanks for the amazing work on the tests.  I downloaded and tested the branch tonight. 

Here are comments from my testing:

1) Choosing between "default" and "custom" is unnecessary. Just present the editable field with the default markers pre-filled. 

2) Error when only one marker is added should be improved: "itemForm:newmarkers: Validation Error: Length is less than allowable minimum of '2'"

I'm happy to help or test more.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java [hunk 1]
```diff
@@ -1274,6 +1275,22 @@ public void setSerializedImageMap(String serializedString) {
...
 	}
 
+  /**
+   * ￼ * for fib questions
+   *
+   * @param param the pair of markers
+   */
+  public void setMarkersPair(String param) {
+    String paramTmp = param;
+    if ((paramTmp == null) || (paramTmp.length() < 2)) paramTmp = "{}";
```
Comment (FerAlvarez95) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java:1285]: We are not allowed to, for example, set just one char, so if we try it, markers are stabilished as default {}. Should we remove de length < 2 check, so we can show the error properly error message?

**Review by FerAlvarez95 — COMMENTED** [review]
Thanks for your comments! Done!

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java (hunk 1 above)
Comment (ottenhoff) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/ItemBean.java:1285]: i think this is okay

**Comment by ottenhoff** [comment]
> Thanks for your comments! I have done both

Okay, I still see the choice for "Fill In The Blank Field Markers" for "default" or "custom". I think there should be no radio button. The instructor should just override the markers if they want. 

I have made tab changes directly to your branch via Github editing.

I also have more small changes that I will make to your branch via a PR

**Comment by FerAlvarez95** [comment]
I think I removed the radio buttons as I removed the properties for default, custom and so. Anyway, thanks for the reviews and changes, I appreciate it.

**Comment by ottenhoff** [comment]
If you agree with my changes, please merge them into your branch

**Review by ottenhoff — APPROVED** [review]

**Pull request merged**